### ISSUE TITLE
在 CPU 和内存都充足的情况下仍无法进入 Epoch 训练

### ISSUE THREAD

**Issue by Bevis-nb** [ultralytics/ultralytics#22291]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Ultralytics 8.3.204  Python-3.13.5 torch-2.8.0+cpu CPU (Intel Xeon CPU E5-2450 v2 @ 2.50GHz)
engine\trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, clos
e_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=coco8.yaml, degrees=0.0, deterministic=
True, device=cpu, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchs
cript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0
=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo12n.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=train5, nbs=64, nms
=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False,
 project=None, rect=False, resume=False, retina_masks=False, save=True, save_conf=False, save_crop=False, save_dir=E:\Projects\test\runs\detect\train5, save
_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.5, seed=0, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True,
 simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=detect, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbos
e=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      6640  ultralytics.nn.modules.block.C3k2            [32, 64, 1, False, 0.25]
  3                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
  4                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]
  5                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
  6                  -1  2    180864  ultralytics.nn.modules.block.A2C2f           [128, 128, 2, True, 4]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  2    689408  ultralytics.nn.modules.block.A2C2f           [256, 256, 2, True, 1]
  9                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 10             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 11                  -1  1     86912  ultralytics.nn.modules.block.A2C2f           [384, 128, 1, False, -1]
 12                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 13             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 14                  -1  1     24000  ultralytics.nn.modules.block.A2C2f           [256, 64, 1, False, -1]
 15                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 16            [-1, 11]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 17                  -1  1     74624  ultralytics.nn.modules.block.A2C2f           [192, 128, 1, False, -1]
 18                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 19             [-1, 8]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 20                  -1  1    378880  ultralytics.nn.modules.block.C3k2            [384, 256, 1, True]
 21        [14, 17, 20]  1    464912  ultralytics.nn.modules.head.Detect           [80, [64, 128, 256]]
YOLOv12n summary: 272 layers, 2,602,288 parameters, 2,602,272 gradients, 6.7 GFLOPs

Transferred 691/691 items from pretrained weights
Freezing layer 'model.21.dfl.conv.weight'
train: Fast image access  (ping: 0.10.0 ms, read: 319.8126.6 MB/s, size: 50.0 KB)
train: Scanning E:\Projects\PtForClassification\datasets\coco8\labels\train.cache... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 493.4Kit/s 0.
train: Scanning E:\Projects\PtForClassification\datasets\coco8\labels\train.cache... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 13.5Kit/s 0.0
s
val: Fast image access  (ping: 0.10.0 ms, read: 408.1179.4 MB/s, size: 54.0 KB)
val: Scanning E:\Projects\PtForClassification\datasets\coco8\labels\val.cache... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 15.5Kit/s 0.0s
后面代码就没有输出，停止了

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bevis-nb，感谢你提交 issue 并使用 Ultralytics 🚀！这是一个自动回复，Ultralytics 工程师也会尽快加入协助你一起排查问题。

我们建议先访问 [Docs](https://docs.ultralytics.com/) 获取快速用法示例和常见问题解答，包括丰富的 [Python](https://docs.ultralytics.com/usage/python/) 与 [CLI](https://docs.ultralytics.com/usage/cli/) 指南。

如果这是一个 🐛 Bug Report，请提供一个可复现的[最小可复现示例（MRE）](https://docs.ultralytics.com/help/minimum-reproducible-example/)以便我们定位问题。为更高效地协助你，请补充：
-操作系统与环境信息（Python/PyTorch版本、CPU/GPU、是否使用虚拟环境或Docker）  
-完整运行命令与配置（例如启动训练的命令、使用的yaml与超参、是否启用val/plots等）  
-最小数据样例或可公开的示例数据路径结构（如你当前的 coco8 子集）  
-完整日志输出（从启动到停止处，建议添加 `verbose=True` 或 CLI 的 `yolo settings` 输出）  
-尝试过的最小化设置结果（例如 `batch=1 workers=0 imgsz=320 epochs=1 device=cpu` 是否仍卡住）  
-确认是否在全新干净环境中重试，或更换到官方示例数据/模型后复现情况

如果这是自定义训练的 ❓ 咨询，请尽可能提供详细信息（数据样例、训练日志等），并确认你遵循了我们的[最佳训练技巧](https://docs.ultralytics.com/guides/model-training-tips/) ✅

加入社区与你的使用场景更快对接：实时交流可前往 [Discord](https://discord.com/invite/ultralytics) 🎧；偏向深入讨论可使用 [Discourse](https://community.ultralytics.com/)；也欢迎在我们的 [Subreddit](https://www.reddit.com/r/Ultralytics/) 分享与交流。

## Upgrade

请先在干净的环境中升级到最新版 `ultralytics`（包含所有[requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)），并在[Python>=3.8](https://www.python.org/) 与 [PyTorch>=1.8](https://pytorch.org/get-started/locally/) 的环境里重试，以核实问题是否已在最新版本中解决：

```bash
pip install -U ultralytics
```

## Environments

YOLO 可在以下已验证、预装依赖（含[CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn)、[Python](https://www.python.org/) 与 [PyTorch](https://pytorch.org/)）的环境中运行：

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

若该徽章为绿色，则所有[Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule)测试当前均通过。CI 会在 macOS、Windows、Ubuntu 上对所有 YOLO[模式](https://docs.ultralytics.com/modes/)与[任务](https://docs.ultralytics.com/tasks/)进行验证，每日与每次提交都会运行。

再次感谢你的反馈！请按上述清单补充 MRE 和环境信息，我们将更快帮助你定位“进入 Epoch 训练前停止输出”的原因 🙏

**Comment by Bevis-nb** [comment]
我就运行了以下代码
from ultralytics import YOLO

# Load a COCO-pretrained YOLO12n model
model = YOLO("yolo12n.pt")

# Train the model on the COCO8 example dataset for 100 epochs
results = model.train(data="coco8.yaml", epochs=100, imgsz=640)

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?
